rfc/0002_scope_manager.md

```diff
@@ -0,0 +1,61 @@
+# 0002: SpanManager
+
+**Current State:** Draft  
+**Author:** [carlosalberto](https://github.com/carlosalberto)
+
+In the OpenTracing specification, under the "Optional API Elements" section, it is mentioned languages may choose to provide utilities to pass an active **Span** around a single process.
+
+Upon many iterations and feedback from several contributors, the Java 0.31 API defined the new **Scope Manager** concept, which is a simple and explicit way to manage the active **Span** at a given call-context point. This document intends to standardize this concept, so other supported languages and platforms can leverage it, with their respective semantical differences.
+
+# Technical background
+
+For any thread, at most one **Span** may be "active". Of course there may be many other **Spans** involved with the thread which are (a) started, (b) not finished, and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child **Span**, or otherwise off of the critical path.
+ 
+It's inconvenient to pass an active **Span** from function to function manually, so OpenTracing should require that every **Tracer** contains a **Scope Manager** that grants access to the active **Span** through a container, called **Scope**.
+
+# Specification Changes
+
+New `ScopeManager` and `Scope` interfaces are added to the specification, and the `Tracer` interface is extended to support creation of `Span`s that are automatically set as the **active** one for the current context.
+
+## ScopeManager
+
+The `ScopeManager` interface allows setting the active `Span` in a call-context storage section, and has the following members:
+
+* **activate**, capability to set the specified `Span` as the active one for the current call-context, returning a `Scope` containing it. A parameter (still to decide whether it will be optional or required) will mark whether the returned `Scope` should, upon deactivation, **finish** the contained `Span`.
+* **active**, the `Scope` containing the current active `Span` if any, or else null/nothing.
+
+## Scope
+
+The `Scope` interface acts as a container of the active `Span` for the current-call context, and has the following members:
+
+* **span**, the contained active `Span` for this call-context. It will never be null/nothing.
+* **close**, marking the end of the active period for the current `Span`, and optionally **finishing** it. Calling it more tan once leads to undefined behavior.
+
+If the language supports some kind of auto finishing statement (such as `try` for Java, or `with` for Python), `Scope` should adhere to such convention. Additionally, `Scope` is not guaranteed to be thread-safe.
+
+## Tracer changes
+
+The `Tracer` interface will be extended with:
+
+* **scope manager**, the `ScopeManager` tracking the active `Span` for this instance.
+* **start active span**, a new behavior for starting a new `Span`, which will be automatically marked as active for the current call-context. It will return a `Scope`.
```

Comment:
This doesn't mention the `finishSpanOnClose` parameter. In C#, we'd like to make the default value `finishSpanOnClose: true` because this is the main use case there. 

My argument for adding a default behavior of `finishSpanOnClose: true` on this method on all languages is that this method creates the scope **and** the span so it is "responsible" for the span. Closing the scope should therefore also finish the scope (unless defined differently by the developer).

Reply:
Agreed - although the mention of such parameter will be done on a more high-level way ;)